hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMClientProtocolServer.java

```diff
@@ -737,17 +807,34 @@ public Pipeline createReplicationPipeline(HddsProtos.ReplicationType type,
   }
 
   @Override
-  public List<Pipeline> listPipelines() {
-    AUDIT.logReadSuccess(
-        buildAuditMessageForSuccess(SCMAction.LIST_PIPELINE, null));
-    return scm.getPipelineManager().getPipelines();
+  public List<Pipeline> listPipelines() throws IOException {
+    try {
+      List<Pipeline> pipelines = scm.getPipelineManager().getPipelines();
+      AUDIT.logReadSuccess(buildAuditMessageForSuccess(
+          SCMAction.LIST_PIPELINE, null));
+      return pipelines;
+    } catch (Exception ex) {
+      AUDIT.logReadFailure(buildAuditMessageForFailure(
+          SCMAction.LIST_PIPELINE, null, ex));
+      throw ex;
+    }
   }
 
   @Override
   public Pipeline getPipeline(HddsProtos.PipelineID pipelineID)
       throws IOException {
-    return scm.getPipelineManager().getPipeline(
-        PipelineID.getFromProtobuf(pipelineID));
+    final Map<String, String> auditMap = Maps.newHashMap();
+    auditMap.put("pipelineID", String.valueOf(pipelineID));
+    try {
+      Pipeline pipeline = scm.getPipelineManager().getPipeline(PipelineID.getFromProtobuf(pipelineID));
```

Comment:
Sorry, I just noticed that this logs `PipelineID` as proto message, which does not look nice in the log:

```
2025-04-15 11:00:56,041 | INFO  | SCMAudit | user=recon/recon@EXAMPLE.COM | ip=172.20.0.2 | op=GET_PIPELINE {pipelineID=id: "84c958af-43c3-42cd-943f-5ab172cfd6b2"
uuid128 {
  mostSigBits: -8878467680604831027
  leastSigBits: -7764387514381510990
}
} | ret=SUCCESS |  
```

Please use `pipelineID.getId()` for audit map value, like it is for `activatePipeline` and other similar methods.